src/main/groovy/com/twcable/grabbit/client/servlets/GrabbitJobServlet.groovy

```diff
@@ -147,6 +147,28 @@ class GrabbitJobServlet extends SlingAllMethodsServlet {
         response.writer.write(new JsonBuilder(jobExecutionIds).toString())
     }
 
+    @Override
+    protected void doPost(SlingHttpServletRequest request, SlingHttpServletResponse response)  {
+        String jobExecutionId = request.getParameter("jobId") ?: ""
+        if(!jobExecutionId.isLong()) {
+            log.warn "Parameter ${jobExecutionId} 'jobId' is invalid"
+            response.status = SC_BAD_REQUEST
+            response.writer.write("Parameter 'jobId' is invalid")
+            return
+        }
+        try {
```

Comment:
Can you please confirm that is the case, because that is contrary to the documentation : http://docs.spring.io/spring-batch/apidocs/org/springframework/batch/core/launch/JobOperator.html#stop-long-

In your scenario, the jobExecutionId would still be present within the job repository; So when Spring Batch queries the repository, it should not return NoSuchJobExecutionException..

In your scenario, I would expect JobExecutionNotRunningException..